_assets/js/netlify/markdownWidgets/collapsible.js

```diff
@@ -0,0 +1,38 @@
+export default function list(engine) {
+  engine.registerTag('collapsible', {
+    parse(tagToken, remainTokens) {
+      this.value = tagToken.args;
+      this.tpls = [];
+      this.liquid.parser
+        .parseStream(remainTokens)
+        .on('template', (tpl) => this.tpls.push(tpl))
+        // note that we cannot use arrow function because we need `this`
+        .on('tag:endcollapsible', function () {
+          this.stop();
+        })
+        .on('end', () => {
+          throw new Error(`tag ${tagToken.getText()} not closed`);
+        })
+        .start();
+    },
```

Comment:
this function is the same across multiple files.

Any way to abstract it?

Reply:
updated!